include/arch/xtensa/arch.h

```diff
@@ -115,17 +115,12 @@ extern void _irq_priority_set(u32_t irq, u32_t prio, u32_t flags);
 #define _ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p) \
 ({ \
 	enum { IRQ = irq_p }; \
-	static struct _isr_table_entry \
-		_CONCAT(_isr_irq, irq_p) \
-		__attribute__ ((used)) \
-		__attribute__ ((section(\
-			STRINGIFY(_CONCAT(.gnu.linkonce.d.isr_irq, irq_p)))\
-		)) = {isr_param_p, isr_p}; \
-	_irq_priority_set(irq_p, priority_p, flags_p); \
+	extern struct _isr_table_entry _sw_isr_table[]; \
+	_sw_isr_table[IRQ].isr = isr_p; \
```

Comment:
I tried using this macro and `gen_isr_tables.py`, but got a `Invaild bfd target` from `objcopy`. `zephyr_prebuilt.elf` is indeed an Xtensa ELF and seems valid, but it beats me why the `xtensa-elf32-elf-objcopy` isn't recognizing it.

I want to clean up a lot of the exception/interrupt handling in the Xtensa port, but there are higher priority items I need to clean up beforehand.

(Also, it might be the case we'll need a `gen_isr_tables.py` specific for Xtensa.)